test/base_test.py

```diff
@@ -0,0 +1,77 @@
+import sys
+from unittest import TestCase
+
+from linode.objects.base import Base, Property
+
+if sys.version_info >= (3, 4):
+    from unittest import mock
+else:
+    import mock
+
+
+class Bar(Base):
+    """
+    A dummy object that can be a child of another object.
+    """
+    api_endpoint = "/bar/{id}"
+    properties = {
+        'id': Property(identifier=True),
+        'value': Property()
+    }
+
+
+class Foo(Base):
+    """
+    A dummy object we will use to test methods of the Base class.
+    """
+    api_endpoint = "/foo/{id}"
+    properties = {
+        'filterable': Property(filterable=True),
+        'id': Property(identifier=True),
+        'something': Property(),
+        'bar': Property(slug_relationship=Bar)
```

Comment:
Below, you're setting 'bar' to `2` - `slug_relationship` is intended to be used for relationships based on strings, like `Linode.region` - this looks more like an `id_relationship` (and by convention, 'bar' would be returned as 'bar_id')